floor_generator/lib/processor/queryable_processor.dart

```diff
@@ -62,40 +77,52 @@ abstract class QueryableProcessor<T extends Queryable> extends Processor<T> {
       orElse: () => null, // whenever field is @ignored
     );
     if (field != null) {
-      final parameterValue = "row['${field.columnName}']";
-      final castedParameterValue = _castParameterValue(
-          parameterElement.type, parameterValue, field.isNullable);
+      final databaseValue = "row['${field.columnName}']";
+
+      String parameterValue;
+
+      if (!parameterElement.type.isDefaultSqlType) {
+        final typeConverter = [...queryableTypeConverters, field.typeConverter]
+            .filterNotNull()
+            .getClosest(parameterElement.type);
+        final castedDatabaseValue =
+            databaseValue.asType(typeConverter.databaseType, field.isNullable);
+
+        parameterValue =
+            '_${typeConverter.name.decapitalize()}.decode($castedDatabaseValue)';
+      } else {
+        parameterValue =
+            databaseValue.asType(parameterElement.type, field.isNullable);
+      }
+
       if (parameterElement.isNamed) {
-        return '$parameterName: $castedParameterValue';
+        return '$parameterName: $parameterValue';
       }
-      return castedParameterValue; // also covers positional parameter
+      return parameterValue; // also covers positional parameter
     } else {
       return null;
     }
   }
+}
 
-  @nonNull
-  String _castParameterValue(
-    final DartType parameterType,
-    final String parameterValue,
-    final bool isNullable,
-  ) {
-    if (parameterType.isDartCoreBool) {
+extension on String {
+  String asType(DartType dartType, bool isNullable) {
+    if (dartType.isDartCoreBool) {
       if (isNullable) {
-        // maps int to bool
-        // if the value is null, return null. If the value is not null, interpret 1 as true and 0 as false.
-        return '$parameterValue == null ? null : ($parameterValue as int) != 0';
+        // if the value is null, return null
+        // if the value is not null, interpret 1 as true and 0 as false
+        return '$this == null ? null : ($this as int) != 0';
       } else {
-        return '($parameterValue as int) != 0'; // maps int to bool
+        return '($this as int) != 0';
       }
-    } else if (parameterType.isDartCoreString) {
-      return '$parameterValue as String';
-    } else if (parameterType.isDartCoreInt) {
-      return '$parameterValue as int';
-    } else if (parameterType.isUint8List) {
-      return '$parameterValue as Uint8List';
+    } else if (dartType.isDartCoreString) {
+      return '$this as String';
+    } else if (dartType.isDartCoreInt) {
+      return '$this as int';
+    } else if (dartType.isUint8List) {
+      return '$this as Uint8List';
     } else {
-      return '$parameterValue as double'; // must be double
+      return '$this as double'; // must be double
```

Comment:
I would rather have a `else if` here and an error if it does not match anything, see #313